### DISCUSSION TITLE
Environment in Revision Links

### DISCUSSION THREAD

**Discussion by QuAzI** [gitextensions/gitextensions#10087] [Q&A]
Is some additional variables for Revision Links possible? Like {host}, {branch}, {parent_branch}, {url} (extept .git in the end)?
As example I want to build PR link for GitLab
```
{url}/-/merge_requests/new?merge_request%5Bsource_branch%5D={branch}&merge_request%5Btarget_branch%5D={parent_branch}
```
Now it is impossible to make global rule and it is impossible to set target branch properly. By default target branch always `master` and sometimes user can forget to change it.

**Comment by gerhardol** [comment]
The only variable defined is %COMMIT_HASH%, defined in GitCommands\ExternalLinks\ExternalLinkFormat.cs

There is a special implementation for GitHub and AzureDevOps templates, also replacing account names etc parsed from the URL. GitUI\CommandsDialogs\SettingsDialog\RevisionLinks\AzureDevopsExternalLinkDefinitionExtractor.cs

**Comment by RussKie** [comment]
Could you please elaborate what you're trying to achieve? Admittedly I haven't used GitLab, and it may have unfamiliar syntax, but I'm finding your example a little bewildering.

**Reply by QuAzI** [reply]
Better integration with available tools. It is possible to add some links to "Related links" section in addition to issue link. As example direct link to CI pipelines for current branch, direct link to create PR/MR with parent branch, direct link to test environment.
Also even issue link can be different on some projects (self-hosted git servers) and now you can't write just one global rule

**Reply by RussKie** [reply]
It does sound non-trivial, perhaps scripts may be a better choice (Settings > Scripts)
